src/module/actor/party/kingdom/model.ts

```diff
@@ -0,0 +1,127 @@
+import { createHTMLElement, fontAwesomeIcon } from "@util";
+import { KingdomCHG, KingdomGovernment, KingdomSchema, KingdomSource } from "./data.ts";
+import { KINGDOM_SCHEMA } from "./schema.ts";
+import type { PartyPF2e } from "../document.ts";
+import { ModelPropsFromSchema } from "types/foundry/common/data/fields.js";
+import { PartyCampaign } from "../types.ts";
+import { ItemType } from "@item/data/index.ts";
+import { FeatGroup } from "@actor/character/feats.ts";
+import { KINGDOM_ABILITIES } from "./values.ts";
+import { resolveKingdomBoosts } from "./helpers.ts";
+import { PartyCampaignSource } from "../data.ts";
+
+const { DataModel } = foundry.abstract;
+
+/** Model for the Kingmaker campaign data type, which represents a Kingdom */
+class KingdomModel extends DataModel<null, KingdomSchema> implements PartyCampaign {
+    declare feats: FeatGroup<PartyPF2e>;
+    declare bonusFeats: FeatGroup<PartyPF2e>;
+
+    static override defineSchema(): KingdomSchema {
+        return KINGDOM_SCHEMA;
+    }
+
+    get extraItemTypes(): ItemType[] {
+        return ["action", "feat"];
+    }
+
+    get level(): number {
+        return this.details.level;
+    }
```

Comment:
Since this already isn't mirroring top-level system data, can we just dispense with the `details` object?

Reply:
We can though I am concerned having a large number of top level properties in a data model subclass. 

I think level is perfect to move to the top level as it should never have been in details to begin with, but I am hesitant on the rest. I'll do so anyways though since data model schema isn't the core concern of this PR and its guaranteed this will be revisited later.

Reply:
Level aside, `details` is for stuff that is more-or-less mechanically insignificant. I believe level being in there for actors is from the original system fork.